sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -1075,8 +1080,8 @@ class GpuMergeAggregateIterator(
           closeOnExcept(new ArrayBuffer[AutoClosableArrayBuffer[SpillableColumnarBatch]]) {
             toAggregateBuckets =>
               var currentSize = 0L
-              while (batchesByBucket.nonEmpty &&
-                batchesByBucket.last.size() + currentSize < targetMergeBatchSize) {
+              while (batchesByBucket.nonEmpty && (toAggregateBuckets.isEmpty ||
```

Comment:
Is it possible that the `toAggregateBuckets` is still empty even after this loop ?
I mean `batchesByBucket` is empty. Since `batchesByBucket` is defined outside of this `RepartitionAggregateIterator` and could be modified during the calls between `hasNext` and `next`.

Reply:
in theory, if you call hasNext() and then manually empty batchesByBucket, and then call next(), you'll get a exception. But in practice there's no such manual manipulation between hasNext() and next()